scripts/modules/b2b-account/account-hierarchy.js

```diff
@@ -0,0 +1,140 @@
+            $(".nested").removeClass("active");
+        },
+        viewAccount: function (e) {
+            //todo: need to implement this method in future.
+            var accountId = e.currentTarget.dataset.mzValue;
+        },
+        changeParentAccount: function (e) {
+            //todo: need to implement this method in future.
+            var accountId = e.currentTarget.dataset.mzValue;
+        }
+    });
+
+    var AccountHierarchyModel = Backbone.MozuModel.extend({
+        mozuType: 'accounthierarchy',
+        initialize: function () {
+            var self = this;
+            self.initApiModel();
+            self.apiGet({ id: require.mozuData('user').accountId }).then(function (response) {
+                self.set("hierarchy", self.processAccountHierarchy(response.data));
+                self.syncApiModel();
+            });
+            self.set("isUserAdmin", self.isUserAdmin());
+        },
+        isUserHavingBehavior: function (behaviorId) {
+            var behaviors = require.mozuData('user').behaviors;
+            if (behaviors) {
+                return behaviors.includes(behaviorId);
+            }
+            return false;
+        },
+        isUserAdmin: function () {
+            //Only admin should have 1000 behavior.
+            return this.isUserHavingBehavior(1000);
+        },
+        isUserPurchaser: function () {
+            //purchaser have 1005 behavior
+            return !this.isUserAdmin() && this.isUserHavingBehavior(1005);
+        },
+        processAccountHierarchy: function (data) {
+            var self = this;
+            if (data.hierarchy) {
+                self.setAccountOnHierarchy(data.hierarchy, data.accounts);
+            }
+            return data.hierarchy;
+        },
+        setAccountOnHierarchy: function (item, accounts, isCurrentAccountChildrens) {
+            var self = this;
+            // isCurrentAccountChildrens will be null/false initially but when current account is logged in users account then set it to true. so we can use this to show menu on children.
+            if (!isCurrentAccountChildrens) {
+                //Check whether this is a current logged in users associated account.
+                isCurrentAccountChildrens = require.mozuData('user').accountId === item.id;
+                //set "View Account" menu on current account.
+                item.isViewAccount = isCurrentAccountChildrens && (self.isUserAdmin() || self.isUserPurchaser());
+            }
+            else {
+                //set menu's on all current accounts children accounts.
+                item.isViewAccount = self.isUserAdmin() || self.isUserPurchaser();
+                item.isChangeParentAccount = self.isUserAdmin();
+            }
+
+            item.account = self.getAccount(item.id, accounts);
+
+            if (item.children) {
+                //do recursive call for children's 
+                self.processAccountHierarchyChildrens(item.children, accounts,
```

Comment:
I would prefer to keep it as is by using processAccountHierarchyChildrens() method because that method has another logic as well, like identify the top-level node of the tree and pass either a single object or list of children's to setAccountOnHierarchy() method.